src/fields/field_fe.cc

```diff
@@ -440,7 +435,7 @@ bool FieldFE<spacedim, Value>::set_time(const TimeStep &time) {
 		bool is_native = (header_query.discretization == OutputTime::DiscreteSpace::NATIVE_DATA);
 		bool boundary;
 		if (is_native || this->interpolation_==DataInterpolation::identic_msh || this->interpolation_==DataInterpolation::equivalent_msh) {
-			n_entities = dh_->mesh()->n_elements();
+			n_entities = ReaderCache::get_mesh(reader_file_)->n_elements(this->boundary_domain_); //dh_->mesh()->n_elements();
```

Comment:
set `n_entities` before the condition as it seems common to both branches. Possibly same holds for the boundary flag.